modules/table/index.js

```diff
@@ -0,0 +1,369 @@
+import Delta from 'quill-delta';
+import { Scope } from 'parchment';
+import Quill from '../../core/quill';
+import Module from '../../core/module';
+import {
+  CellLine,
+  TableCell,
+  TableRow,
+  TableBody,
+  TableContainer,
+  tableId,
+  TableHeaderCell,
+  TableHeaderRow,
+  TableHeader,
+  HeaderCellLine,
+  TABLE_TAGS,
+} from '../../formats/table';
+import isDefined from '../../utils/is_defined';
+import { deltaEndsWith, applyFormat } from '../clipboard';
+import makeTableArrowHandler from './utils/make_table_arrow_handler';
+
+const ELEMENT_NODE = 1;
+
+class Table extends Module {
+  static register() {
+    Quill.register(CellLine, true);
+    Quill.register(HeaderCellLine, true);
+    Quill.register(TableHeaderCell, true);
+    Quill.register(TableCell, true);
+    Quill.register(TableHeaderRow, true);
+    Quill.register(TableRow, true);
+    Quill.register(TableBody, true);
+    Quill.register(TableHeader, true);
+    Quill.register(TableContainer, true);
+  }
+
+  constructor(...args) {
+    super(...args);
+
+    this.integrateClipboard();
+    this.addKeyboardHandlers();
+
+    this.listenBalanceCells();
+  }
+
+  integrateClipboard() {
+    this.quill.clipboard.addTableBlot(CellLine.blotName);
+    this.quill.clipboard.addTableBlot(TableHeaderCell.blotName);
+    this.quill.clipboard.addMatcher('td, th', matchCell);
+    this.quill.clipboard.addMatcher(ELEMENT_NODE, matchDimensions);
+  }
+
+  addKeyboardHandlers() {
+    const bindings = Table.keyboardBindings;
+    Object.keys(bindings).forEach(name => {
+      if (bindings[name]) {
+        this.quill.keyboard.addBinding(bindings[name]);
+      }
+    });
```

Comment:
mb this way?
```
    Object.values(bindings).forEach(binding => {
      if (binding) {
        this.quill.keyboard.addBinding(binding);
      }
    });
```

Reply:
IE11 :( Some of the changes we want to provide privately in 21.1